tekton/pipeline_run_test.go

```diff
@@ -212,6 +212,35 @@ var _ = Describe("PipelineRun", func() {
 			Expect(releasePipelineRun.Spec.TaskRunTemplate.ServiceAccountName).To(Equal(serviceAccountName))
 		})
 
+		It("can add the taskGitRevision parameter to the PipelineRun object when using a git resolver", func() {
+			pipelineRef := &tektonutils.PipelineRef{
+				Resolver: "git",
+				Params: []tektonutils.Param{
+					{
+						Name:  "revision",
+						Value: "my-revision",
+					},
+				},
+			}
+			releasePipelineRun.WithTaskGitRevisionParameter(pipelineRef)
+			Expect(releasePipelineRun.Spec.Params[0].Name).To(Equal("taskGitRevision"))
+			Expect(releasePipelineRun.Spec.Params[0].Value.StringVal).To(Equal("my-revision"))
+		})
+
+		It("does not add the taskGitRevision parameter to the PipelineRun object when using a bundles resolver", func() {
+			pipelineRef := &tektonutils.PipelineRef{
+				Resolver: "bundles",
+				Params: []tektonutils.Param{
+					{
+						Name:  "revision",
```

Comment:
I don't really mind, but maybe it could a more realistic example of a bundle resolver (which typically doesn't have a "revision"). This wouldn't affect the code path taken anyway, since the first `if` that checks if the resolver is git will not pass. Just a thought.

For example:
```go
Params: []tektonutils.Param{
					{
						Name:  "bundle",
						Value: "my-bundle",
					},
					{
						Name:  "name",
						Value: "my-name",
					},
					{
						Name:  "kind",
						Value: "my-kind",
					},
				},
```

Or the params could even be empty.

Reply:
We're extracting the revision parameter, so my goal was to have a resolver with the revision parameter there to ensure it is not still extracted. Maybe I shouldn't have use `bundles` resolver here. Would it help if I changed it to like `Resolver: "my-custom-resolver"`?

Reply:
OK, makes sense.

> Would it help if I changed it to like Resolver: "my-custom-resolver"?

No, it's probably fine, thanks.